### PR TITLE
Issue-29302- Graphql schema change

### LINKED ISSUE #29302: Input type of customizable option doesn't get returned through GraphQL
### Preconditions

Github branches `magento2/2.4-develop` and `architecture/master`

### Steps to reproduce

The GraphQL type `SelectedCustomizableOption` is used in multiple places across the Magento GraphQL modules to get the currently selected customizable option of a product through GraphQL. There are various field types for this such as text, textarea, select, multiselect, checkbox, etc. Currently, there's no way for a frontend using GraphQL to know how to render the SelectedCustomizableOption. Yes, the format varies according to different types, but the format is the same for input types of the same category. I believe there's no way to differentiate between a text and a textarea of `SelectedCustomizableOption`. Same with the different select input types. 

The associated resolver class _does_ return the 'type of input' from Magento (ref: https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/QuoteGraphQl/Model/CartItem/DataProvider/CustomizableOption.php#L60), but the associated GraphQL schema does not define a field for the same (ref: https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/QuoteGraphQl/etc/schema.graphqls#L346) and hence the 'type of input' doesn't get returned. 

Upon some searching, the GraphQL coverage docs did have the 'type of input' as a field in `SelectedCustomizableOption` a couple of months ago (ref: https://github.com/magento/architecture/blob/673438109bbf63d819e96c373ef7622206ff7f9b/design-documents/graph-ql/coverage/add-items-to-cart/AddSimpleProductToCart.graphqls#L59) until the new coverage docs (ref: https://github.com/magento/architecture/blob/master/design-documents/graph-ql/coverage/Cart.graphqls#L114) was matched according to the current GraphQL schema and was thus, removed. 

### Expected result

Consistency.

Is the field `type` required to render the input type on the frontend?
-  If yes, we need to add the field back to the schema.
-  If not, we need to remove it from the resolver for consistency and to avoid confusion for developers in the future.

### Actual result

The field `type` was removed from the QuoteGraphQl module in this commit: https://github.com/magento/magento2/commit/1315577e2099637207f02deec48607d81f7cde46#diff-795a33fde881f18aba5165a5a8c7513fL317 and from the architecture coverage docs a couple months ago and now the code is inconsistent and causes confusion for developers.

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [x] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

### PR DESCRIPTION
**Pull request by dineshvb**
## Description (*)
- Added the **"type"** field in the schema of QuoteGraphQl's **"SelectedCustomizableOption"** type.
- This field is used to maintain **consistency** between the Resolver class and Graphql schema.
- Also, the type field returns the input types like Checkbox, Radio Button, etc., which helps the **frontend** developers to **render** the PDP page if required.

### Fixed Issues (if relevant)

1. Fixes magento/magento2#29302

### Manual testing scenarios (*)

1. ... Added a few **customizable options** to a few products in local admin and checked it via **Altair** GraphQL client

**Sample Request :** 

```
{
  cart(cart_id: String!)
  {
    items
    {
      id
      quantity
      product{
        name
      }
      ... on SimpleCartItem{
       alias1: customizable_options{
          id
          is_required
          label
          sort_order
          type
        }
      }
      
      ... on DownloadableCartItem{
      alias2: customizable_options{
          id
          is_required
          label
          sort_order
          type
         
        }
      }
    }
  }
}
```

**Sample Response:**

```
{
  "data": {
    "cart": {
      "items": [
        {
          "id": "7",
          "quantity": 1,
          "product": {
            "name": "paint"
          },
          "alias1": [
            {
              "id": 1,
              "is_required": true,
              "label": "size",
              "sort_order": 1,
              "type": "checkbox"
            }
          ]
        },
        {
          "id": "12",
          "quantity": 1,
          "product": {
            "name": "Music CDs"
          },
          "alias2": [
            {
              "id": 3,
              "is_required": true,
              "label": "track",
              "sort_order": 1,
              "type": "radio"
            }
          ]
        },
        {
          "id": "13",
          "quantity": 2,
          "product": {
            "name": "paint"
          },
          "alias1": [
            {
              "id": 1,
              "is_required": true,
              "label": "size",
              "sort_order": 1,
              "type": "checkbox"
            }
          ]
        }
      ]
    }
  }
}
```

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by swnsma** [comment]
Hi @dineshvb,
Could you sign Adobe CLA before we can proceed with reviewing your Pull Request?

**Comment by nrkapoor** [comment]
@magento run all tests

**Comment by dineshvb** [comment]
@swnsma Since there were a few existing static failures with the schema, how should I proceed? Thank you

**Comment by dineshvb** [comment]
@swnsma Can you please look through the static tests that were failing?

**Comment by dineshvb** [comment]
@keharper Since there were a few existing static failures with the schema, how should I proceed? Thank you

**Comment by swnsma** [comment]
Hi @dineshvb,
Sorry for long response.

Hi @sidolov,
What it the better approach in such situation?
I believe that changing names of already available fields will cause a BC break.

**Comment by keharper** [comment]
@dineshvb I'm just the documentation guy, so I can't really help with this. All I can say is please open a devdocs PR if your changes are merged.

**Comment by sidolov** [comment]
@dineshvb , @swnsma I think we should proceed with changes since the fields already exist in schema

**Review by swnsma — APPROVED** [review]
Failed tests approved.
Current violations do not related to current PR.
Moreover, fix of violations will introduce BC break.

**Comment by magento-engcom-team** [comment]
Hi @swnsma, thank you for the review.
[ENGCOM-8463](https://jira.corp.magento.com/browse/ENGCOM-8463) has been created to process this Pull Request
:eight_spoked_asterisk: @swnsma, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by dineshvb** [comment]
> Hi @swnsma, thank you for the review.
> [ENGCOM-8463](https://jira.corp.magento.com/browse/ENGCOM-8463) has been created to process this Pull Request
>  @swnsma, could you please add one of the following labels to the Pull Request?
> 
> Label	Description
> `Auto-Tests: Covered`	All changes in Pull Request is covered by auto-tests
> `Auto-Tests: Not Covered`	Changes in Pull Request requires coverage by auto-tests
> `Auto-Tests: Not Required`	Changes in Pull Request does not require coverage by auto-tests

@swnsma can you add a label to this?

**Review by swnsma — CHANGES_REQUESTED** [review]
Hi @dineshvb,

According to Magento Definition of Done, all new functionality should be covered with auto-tests.
May I ask you to add test coverage for newly introduced behaviour?
You could extend already available tests to also check type field (e.g. \Magento\GraphQl\Quote\EditQuoteItemWithCustomOptionsTest::testOptionsSetPersistsOnQtyChange)

File: app/code/Magento/QuoteGraphQl/etc/schema.graphqls [hunk 1]
```diff
@@ -350,6 +350,7 @@ type CartItemPrices {
 type SelectedCustomizableOption {
     id: Int!
     label: String!
+    type: String!
```
Comment (cpartica) [inline app/code/Magento/QuoteGraphQl/etc/schema.graphqls:357]: this needs a simple test

**Review by cpartica — CHANGES_REQUESTED** [review]

File: app/code/Magento/QuoteGraphQl/etc/schema.graphqls (hunk 1 above)
Comment (dineshvb) [reply app/code/Magento/QuoteGraphQl/etc/schema.graphqls:357]: Hi @cpartica , I have some issues while installing PHPUnit in my system. That's why I haven't issued the PR yet. I will soon resolve and give a PR

**Comment by swnsma** [comment]
Hi @dineshvb,
It looks like you have commited your Magento Commerce symlinks. 
Could you reset these changes?

**Comment by dineshvb** [comment]
@magento run all tests

**Comment by dineshvb** [comment]
@magento run all tests

**Comment by dineshvb** [comment]
The test failures here are not related to my changes which come under the WEB API graphql tests that were successful

**Comment by swnsma** [comment]
@magento run Functional Tests B2B, Functional Tests CE, Functional Tests EE

**Comment by swnsma** [comment]
@magento run Functional Tests B2B, Functional Tests EE

**Comment by dineshvb** [comment]
@magento run Functional Tests B2B, Functional Tests EE

**Comment by swnsma** [comment]
@magento run Functional Tests B2B, Functional Tests EE

**Comment by swnsma** [comment]
@magento run Functional Tests B2B, Functional Tests EE

**Comment by dineshvb** [comment]
@magento run Functional Tests B2B

**Review by swnsma — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @swnsma, thank you for the review.
[ENGCOM-8463](https://jira.corp.magento.com/browse/ENGCOM-8463) has been created to process this Pull Request

**Pull request merged**